### PR TITLE
feat: use location instead of get header

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: ionoscloud/resource_server.go [hunk 1]
```diff
@@ -648,7 +648,11 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
...
 
 	// Wait, catching any errors
-	_, errState := cloudapi.GetStateChangeConf(meta, d, apiResponse.Header.Get("Location"), schema.TimeoutCreate).WaitForStateContext(ctx)
+	loc, err := apiResponse.Location()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	_, errState := cloudapi.GetStateChangeConf(meta, d, loc.String(), schema.TimeoutCreate).WaitForStateContext(ctx)
 	if errState != nil {
 		if cloudapi.IsRequestFailed(err) {
```
Comment (cbeti-ionos) [inline ionoscloud/resource_server.go:657]: redeclaring err in `loc, err := apiResponse.Location()` might affect this, 
but I am unsure what the point of this is in the first place

previously you couldn't reach this point if `err` was not `nil` after the create request, shouldn't it have been `errState`?

File: ionoscloud/resource_server.go [hunk 2]
```diff
@@ -964,7 +972,11 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
...
 		// Wait, catching any errors
-		_, errState := cloudapi.GetStateChangeConf(meta, d, apiResponse.Header.Get("Location"), schema.TimeoutUpdate).WaitForStateContext(ctx)
+		loc, err := apiResponse.Location()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		_, errState := cloudapi.GetStateChangeConf(meta, d, loc.String(), schema.TimeoutUpdate).WaitForStateContext(ctx)
 		if errState != nil {
 			diags := diag.FromErr(errState)
 			return diags
```
Comment (cbeti-ionos) [inline ionoscloud/resource_server.go:982]: this is a lot of repetition that could be refactored to it's own function

File: ionoscloud/resource_server.go (hunk 1 above)
Comment (cristiGuranIonos) [reply ionoscloud/resource_server.go:657]: Yes, this should be errState.

File: ionoscloud/resource_server.go (hunk 2 above)
Comment (cristiGuranIonos) [reply ionoscloud/resource_server.go:986]: which part? getting the location and checking state change?

**Review by cbeti-ionos — APPROVED** [review]

**Pull request merged**